src/advancedKubectl/content.md

```diff
@@ -0,0 +1,1108 @@
+    },
+    {
+      "image": "kubernetes-internals-3.svg",
+      "description": "The ReplicaSet controller creates a Pod definition for each replica of the ReplicaSet (according to the Pod template in the ReplicaSet definition) and saves them in the storage backend."
+    },
+    {
+      "image": "kubernetes-internals-4.svg",
+      "description": "This triggers the scheduler who watches for Pods that have not yet been assigned to a worker node."
+    },
+    {
+      "image": "kubernetes-internals-5.svg",
+      "description": "The scheduler chooses a suitable worker node for each Pod and adds this information to the Pod definitions in the storage backend."
+    },
+    {
+      "image": "kubernetes-internals-6.svg",
+      "description": "This triggers the kubelet on the worker node that the Pods have been scheduled to, who watches for Pods that have been scheduled to its worker node."
+    },
+    {
+      "image": "kubernetes-internals-7.svg",
+      "description": "The kubelet reads the Pod definitions from the storage backend, downloads the required container images, and runs the containers via Docker (or another container runtime) on the worker node."
+    }
+  ]
+}
+```
+
+The API request to the *create ReplicaSet* endpoint is handled by the **API server**. The API server authenticates the request and saves your ReplicaSet resource definition in the storage backend.
+
+This event triggers the **ReplicaSet controller**, which is a sub-process of the **controller manager**. The ReplicaSet controller watches for creations, updates, and deletions of ReplicaSet resources in the storage backend, and gets notified by an event when this happens. 
+
+The job of the ReplicaSet controller is to make sure that the required number of replica Pods of a ReplicaSet exists. In our example, no Pods exist yet, so the ReplicaSet controller creates these Pod definitions (according to the Pod template in the ReplicaSet definition) and saves them in the storage backend.
+
+The creation of the new Pods triggers the **scheduler**, who watches for Pod definitions that are not yet scheduled to a worker node. The scheduler chooses a suitable worker node for each Pod and updates the Pod definitions in the storage backend with this information.
+
+> Note that up to this point, no workload code is being run anywhere in the cluster. All that has been done so far is creating and updating resources in the storage backend on the master node.
+
+This event triggers the **kubelets** who watch for Pods that are scheduled to their worker nodes. The kubelet of the worker node your ReplicaSet Pods have been scheduled to downloads the container images for these Pods (if not already present on the machine) and runs them with the configured container runtime (which may be Docker or another container runtime).
+
+At this point, finally, your ReplicaSet application is running!
```

Comment:
**Genuine question.** The content is obviously duplicated: once in the slideshow and the other after it.
I have the exact same problem in the Academy and I decided to revert the order: first I explain, then I reiterate using the slideshow. You went for the opposite, is there a reason behind it?

Reply:
I used this order to interrupt the text so that the readers have something visual to look at before they have to read a lot of text. The other way would be possible too, but the whole section might look less balanced. I think in books you can also find both approaches, first the figure then the explanations, and first the explanations then a figure as a summary.